lib/levelup.js

```diff
@@ -65,9 +65,17 @@ function LevelUP (location, options, callback) {
     options  = {}
   }
 
-  if ((!options || typeof options.db != 'function') && typeof location != 'string') {
+  if (!options && typeof options.db != 'function') {
```

Comment:
Going through what you did in this PR and it seems to me that this line is wrong, shouldn't it be

```js
if (!options || typeof options.db !== 'function') {}
```